python/lsst/ts/standardscripts/maintel/laser_tracker/align.py

```
@@ -0,0 +1,213 @@
+        - "M2 Hexapod aligned with laser tracker.": M2 aligned.
+        - "Camera Hexapod aligned with laser tracker.": M2 aligned.
+
+        Raises
+        ------
+        ValueError
+            If target is not 1 or 3.
+
+        RuntimeError
+            If not aligned after max_iter iterations.
+        """
+
+        # Set function to align target.
+        if target == 1:
+            align_func = self.mtcs.offset_m2_hexapod()
+        elif target == 3:
+            align_func = self.mtcs.offset_camera_hexapod()
+        else:
+            raise ValueError(f"Invalid target: {target}")
+
+        for n_iter in range(self.max_iter):
+            await self.laser_tracker.rem.lasertracker_1.evt_offsetsPublish.flush()
+            await self.laser_tracker.rem.lasertracker_1.cmd_align.set_start(
+                target=target,
+                timeout=self.timeout_align,
+            )
+            offset = (
+                await self.laser_tracker.rem.lasertracker_1.evt_offsetsPublish.next(
+                    flush=False, timeout=self.timeout_std
+                )
+            )
+
+            corrections = [
+                offset.dX * 1e6 if abs(offset.dX) > self.tolerance_m else 0.0,
+                offset.dY * 1e6 if abs(offset.dY) > self.tolerance_m else 0.0,
+                offset.dZ * 1e6 if abs(offset.dZ) > self.tolerance_m else 0.0,
+                offset.dRX if abs(offset.dRX) > self.tolerance_deg else 0.0,
+                offset.dRY if abs(offset.dRX) > self.tolerance_deg else 0.0,
+            ]
+
+            if any(corrections):
+                self.log.info(
+                    f"[{n_iter+1:02d}:{self.max_iter:02d}]: Applying corrections: {corrections}"
+                )
+                await align_func(*-corrections)
+
+            else:
+                self.log.info(
+                    f"[{n_iter+1:02d}:{self.max_iter:02d}]: Corrections completed."
+                )
+                break
```

Instead of breaking, consider returning, than you can simply `raise  RuntimeError` if it gets past the for loop.

---

Done